queue_processors/autoreduction_processor/post_process_admin.py

```diff
@@ -206,12 +214,62 @@ def specify_instrument_directories(self,
         # Specify directories where autoreduction output will go
         return temporary_directory + instrument_output_directory
 
-    def reduction_started(self):
-        """Log and update AMQ message to reduction started"""
-        logger.debug("Calling: %s\n%s",
-                     ACTIVEMQ_SETTINGS.reduction_started,
-                     self.message.serialize(limit_reduction_script=True))
-        self.client.send(ACTIVEMQ_SETTINGS.reduction_started, self.message)
+    @staticmethod
+    def write_and_readability_checks(directory_list, read_write):
+        """
+        Check a list of N directories for write/readability
+        :param directory_list: (list) directory list
+        :param read_write: (str) Read=R, Write=W
+        :return Error (Exception or ValueError) when something goes wrong
+        """
+        read_write = read_write.upper()
+        read_write_map = {"R": "read", "W": "write"}
+        try:
+            read_write_map[read_write]
+        except KeyError:
+            raise KeyError("Invalid read or write input: %s read_write argument must be either"
+                           " 'R' or 'W'" % read_write)
+
+        for location in directory_list:
+            if not os.access(location, getattr(sys.modules[os.__name__], f"{read_write}_OK")):
+                if not os.access(location, os.F_OK):
```

Comment:
Do we need to test if the directory exists after it fails the read/write permission check fails? 
I believe the only reason the directory wouldn't exist is if it was deleted in the time between it was created and here, or it failed to be created. In the second scenario an `OSError` would have already been raised, so we would be aware of this situation already. Alternatively if it had been deleted, it wouldnt be readable/writable this would fail, the reduction would be aborted and it would be recreated on the next run anyway.

Reply:
The purpose of this PR is to not inherently change the processes already taking place, but to simply move processes outside of reduce to first add unit tests before considering the changes which should take place to clean up the script as a whole.
I do however still believe we should test this so we can verbose in understanding the error. If the file exists, that doesn’t necessarily mean that we will have read or write access to the file. 
If you added a command to travis.yml to cat the logs produced by logs/autoreduction_processor.log, you will see that this method is used regularly to create paths that don’t exist when running end-to-end tests. 
If the directory does not exist, we should  try to make it instead of raising an OSERROR and trying the whole reduction again in 6 hours time.